examples/vault-auto-unseal/user-data-vault.sh

```diff
@@ -22,3 +55,32 @@ readonly VAULT_TLS_KEY_FILE="/opt/vault/tls/vault.key.pem"
   --enable-auto-unseal \
   --auto-unseal-kms-key-id "${kms_key_id}" \
   --auto-unseal-kms-key-region "${aws_region}"
+
+# Initializes the vault server
+# Retries this a number of times because run-vault is running on the background and
+# we need to wait for this to finish.
+# This userdata file is present in all instances in the cluster, but the code below
+# will only succeed in the first one that runs it, the others will fail with "the server has
+# already been initialized" and the script will be interrupted. This is not an
+# issue, and, for the purpose of this example, it is simpler than reorganizing the
+# code in such a way that the nodes receive different user data files.
```

Comment:
Hmmm... but what about production use cases? Should we tell users to run `init` manually against one node? Or perhaps better, should we detect the "already initialized" error and ignore it based on the assumption one of the other nodes already did the initialization? I'd rather avoid the script exiting with an error, as that will get very confusing to debug (e.g., if later on we add something after this line that we expect to run, but now doesn't on 2 out of 3 nodes).

Reply:
Following discussion with Josh, we decided to move this bit from the user data to the go tests since it should be manually done by the operator one time. This would then eliminate this issue. But I will leave it as a comment in the user data.

Reply:
Should it still be a manual operation even if you have auto-unseal and could therefore boot up a cluster completely automatically?

Reply:
That is a fair point. But then we have to make an informed decision to suggest on how to handle the root token in production, dont we?

Reply:
Oh, I totally forgot that's when you get a root token. Yea, never mind, best that's handled manually.

Reply:
ok 👍